### PR TITLE
wmma: refactor to remove wmma_func and create TC funcs as needed

### PR DESCRIPTION
**Pull request by flammit**

### CODE REVIEW

File: extra/gemm/simple_matmul.py [hunk 1]
```diff
@@ -1,10 +1,12 @@
 import numpy as np
 from tinygrad.helpers import getenv
 from tinygrad import dtypes, Tensor
-dtype_in = dtypes.half if getenv("HALF") else dtypes.float
-acc_dtype = dtypes.half if getenv("ACC_HALF") else None
+dtype_in = dtypes.half if getenv("HALF") else dtypes.bfloat16 if getenv("BFLOAT16") else dtypes.float
```
Comment (chenyuxyz) [inline extra/gemm/simple_matmul.py:4]: can you add bf16 to ci benchmark?

**Review by chenyuxyz — COMMENTED** [review]
i will review this tomorrow, quite excited about it as it reduces lines and adds capabilities!

also need to sync to master

File: extra/gemm/simple_matmul.py (hunk 1 above)
Comment (flammit) [reply extra/gemm/simple_matmul.py:4]: done!

File: tinygrad/renderer/cstyle.py [hunk 2]
```diff
@@ -242,17 +247,24 @@ class CUDALanguage(CStyleLanguage):
   type_map = {dtypes.bfloat16: "nv_bfloat16"}
 
   def render_kernel(self, function_name, kernel, bufs, uops, prefix=None):
+    # TODO: why is dtypes.bfloat16.name == "__bf16"? would be easier not override dtypes.name
```
Comment (chenyuxyz) [inline tinygrad/renderer/cstyle.py:250]: i think `__bf16` works for clang

**Review by chenyuxyz — COMMENTED** [review]
nice work! looks good

I would suggest next time splitting the make dtype change from the wmma change and make two prs. Easier to debug small individual prs in case we find a regression,

**Pull request merged**